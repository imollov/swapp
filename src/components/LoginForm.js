import React, { Component } from 'react'
import Row from './Row'

class LoginForm extends Component {
  handleTextChange = ({ target: { name, value } }) => {
    this.props.onInputChange({ [name]: value })
  }

  handleSubmit = () => this.props.onSubmit()

  render() {
    const { username, password, className } = this.props
    return (
      <div className={className}>
        <Row>
          <input
            type="text"
            name="username"
            value={username}
            onChange={this.handleTextChange}
          />
        </Row>
        <Row>
          <input
            type="password"
            name="password"
            value={password}
            onChange={this.handleTextChange}
          />
        </Row>
        <Row>
          <button onClick={this.handleSubmit}>Login</button>
        </Row>
      </div>
    )
  }
}

export default LoginForm
