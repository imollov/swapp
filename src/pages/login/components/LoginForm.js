import React, { useState } from 'react'
import { Box, Button, Text } from 'rebass/styled-components'
import StyledInput from './StyledInput'

export default ({ onSubmit, errorMsg, ...rest }) => {
  const [state, setState] = useState({
    email: '',
    password: '',
  })

  const handleInputChange = ({ target: { value, name } }) =>
    setState({ ...state, [name]: value })

  const handleSubmit = e => {
    e.preventDefault()
    onSubmit(state)
  }

  return (
    <Box {...rest} p={4} as="form" onSubmit={handleSubmit}>
      {errorMsg && (
        <Box>
          <Text variant="error">{errorMsg}</Text>
        </Box>
      )}
      <Box my={3}>
        <StyledInput
          id="email"
          name="email"
          placeholder="email"
          value={state.email}
          onChange={handleInputChange}
        />
      </Box>
      <Box my={3}>
        <StyledInput
          id="password"
          name="password"
          type="password"
          placeholder="password"
          value={state.password}
          onChange={handleInputChange}
        />
      </Box>
      <Box textAlign="right">
        <Button variant="primary">Login</Button>
      </Box>
    </Box>
  )
}
