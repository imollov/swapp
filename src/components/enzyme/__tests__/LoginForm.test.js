import React from 'react'
import { shallow } from 'enzyme'

import LoginForm from '../../LoginForm'

describe('LoginForm Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(<LoginForm />)

    expect(wrapper).toExist()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(
      <LoginForm
        username="demo"
        password="qwerty"
        onInputChange={() => {}}
        onSubmit={() => {}}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot with error message', () => {
    const wrapper = shallow(
      <LoginForm
        username="demo"
        password="qwerty"
        onInputChange={() => {}}
        onSubmit={() => {}}
        errorMessage="Invalid credentials!"
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should propagate input change', () => {
    const onInputChangeMock = jest.fn()
    const wrapper = shallow(
      <LoginForm
        username="demo"
        password="qwerty"
        onInputChange={onInputChangeMock}
        onSubmit={() => {}}
      />
    )

    expect(onInputChangeMock).not.toHaveBeenCalled()

    const usernameInput = wrapper.find('input[name="username"]')
    usernameInput.simulate('change', {
      target: { name: 'username', value: 'test' }
    })

    expect(onInputChangeMock).toHaveBeenCalledWith({
      username: 'test'
    })
  })

  it('should propagate submit', () => {
    const onSubmitMock = jest.fn()
    const wrapper = shallow(
      <LoginForm
        username="demo"
        password="qwerty"
        onInputChange={() => {}}
        onSubmit={onSubmitMock}
      />
    )

    expect(onSubmitMock).not.toHaveBeenCalled()

    const submitButton = wrapper.findWhere(
      el => el.text() === 'Login' && el.type() === 'button'
    )
    submitButton.simulate('click')

    expect(onSubmitMock).toHaveBeenCalled()
  })
})
