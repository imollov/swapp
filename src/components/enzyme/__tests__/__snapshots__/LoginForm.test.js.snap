// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should match snapshot 1`] = `
<div>
  <Row>
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value="demo"
    />
  </Row>
  <Row>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="qwerty"
    />
  </Row>
  <Row>
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </Row>
</div>
`;

exports[`LoginForm Component should match snapshot with error message 1`] = `
<div>
  <Row>
    <span>
      Invalid credentials!
    </span>
  </Row>
  <Row>
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value="demo"
    />
  </Row>
  <Row>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="qwerty"
    />
  </Row>
  <Row>
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </Row>
</div>
`;
